lib/web_of_science/process_records.rb

```diff
@@ -0,0 +1,134 @@
+module WebOfScience
+
+  # This class complements the WebOfScience::Harvester
+  # Process records retrieved by any means; this is a progressive filtering of the harvested records to identify
+  # those records that should create a new Publication-PubHash, PublicationIdentifier(s) and Contribution(s).
+  class ProcessRecords
+
+    # @param author [Author]
+    # @param records [WebOfScience::Records]
+    def initialize(author, records)
+      raise(ArgumentError, 'author must be an Author') unless author.is_a? Author
+      raise(ArgumentError, 'records must be an WebOfScience::Records') unless records.is_a? WebOfScience::Records
+      @author = author
+      @records = records
+    end
+
+    # @return [Array<String>] WosUIDs that create a new Publication
+    def execute
+      return [] if records.count.zero?
+      create_publications
+      # TODO: batch harvest PubMed data for WOS records with a PMID?
+    rescue StandardError => err
+      logger.error(err.inspect)
+      []
+    end
+
+    private
+
+      attr_reader :author
+      attr_reader :records
+
+      # ----
+      # Record filters and data flow steps
+
+      # @return [Array<String>] WosUIDs that create a new Publication
+      def create_publications
```

Comment:
i still find this set of method names not totally clear as to what they are intending to do (I know now because we discussed it, but I can see it being unclear for someone else who comes to the code) .... but don't have better suggestions, I don't think we should dwell on it here but maybe iterate on it later as we continue to work on this class

Reply:
OK, one interation is not enough it seems, but in the absence of better ideas let's move on with this for now.